### DISCUSSION TITLE
How should I define the dataset path?

### DISCUSSION THREAD

**Discussion by BugMaker2002** [Dylan-H-Wang/SLF-RPM#7] [Q&A] [answered]
The UBFC-rPPG dataset I download from the Internet like this: 
![image](https://github.com/Dylan-H-Wang/SLF-RPM/assets/110211332/7a56c81c-52f2-4b1e-bd82-d3e8ea1c9e4b)
So the data path should be "/UBFC_rPPG/dataset2", I run the code using this command:

python main.py --gpu 0 --epochs 50 --batch_size 12 --lr 1e-4 \
	--n_dim 2048 --temperature 0.1 \
	--dataset_name "ubfc-rppg" \
	--dataset_dir "/share2/data/zhouwenqing/UBFC_rPPG/dataset2" \
	--workers 4 --vid_frame 150 --clip_frame 30 --roi_list 0 1 2 3 4 5 6 --stride_list 1 2 3 4 5 \
	--log_dir "./logs/ubfc/train" \
	--model_depth 18

But it made a mistake:
![image](https://github.com/Dylan-H-Wang/SLF-RPM/assets/110211332/6b073f92-7086-4610-a391-ed0724cabddf)
I have taken a closer look at the code of your UBFCDataset class, but I find that the data set path your code handles does not seem to be the original UBFC-rPPG data set. Did you make any changes to the original UBFC-rPPG dataset directory format?

**Comment by Dylan-H-Wang** [comment, marked as answer]
Hi,
You need to preprocess the original dataset before using `UBFCDataset`. Please check codes here
https://github.com/Dylan-H-Wang/SLF-RPM/blob/298b01fc47d06506b71eda220e4e231e58f83a66/utils/utils.py#L311-L396

Cheers
